PhysicsTools/PatAlgos/python/tools/jetTools.py

```diff
@@ -634,21 +634,21 @@ def setupBTagging(process, jetSource, pfCandidates, explicitJTA, pvSource, svSou
             if btagInfo == 'pfDeepBoostedJetTagInfos':
                 if pfCandidates.value() == 'packedPFCandidates':
                     jetSrcName = jetSource.value().lower()
-                    if 'slimmed' in jetSrcName or 'updated' in jetSrcName or 'packed' in jetSrcName:
-                        # case 1: update jets whose daughters are PackedCandidates, e.g., slimmedJetsAK8, packedPatJetsAK8, etc.
```

Comment:
@slava77 I thought about this carefully again and I realized that I indeed missed one case, i.e., if analyzers recluster Puppi jets from MiniAOD and want to add these taggers. The updated version should cover this case. The analyzers will have to specify explicitly that the jet daughters are puppi weighted in this case (by adding 'WithPuppiDaughters' in the postfix when calling updateJetCollection).

Still, I think the ideal solution would be to infer whether the packedCandidate are puppi weighted or not from the code, rather than relying on some postfix string. However I am not sure if that is possible.